kinto_fxa/scripts/process_account_events.py

```diff
@@ -0,0 +1,122 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+"""Script to process account-related events from an SQS queue.
+
+This script polls an SQS queue for events indicating activity on an upstream
+account, as documented here:
+
+  https://github.com/mozilla/fxa-auth-server/blob/master/docs/service_notifications.md
+
+The following event types are currently supported:
+
+  * "delete": the account was deleted; we delete their default bucket
+    to comply with GDPR. Note that this won't be sufficient in
+    applications where users can store data in locations besides the
+    default bucket.
+
+Note that this script may not be necessary in all applications of
+kinto-fxa.
+
+"""
+
+import itertools
+import json
+import logging
+import uuid
+
+import boto3
+from ec2_metadata import ec2_metadata
+from kinto.core.utils import hmac_digest
+import transaction as current_transaction
+
+logger = logging.getLogger(__name__)
+
+
+def get_default_bucket_id(config, uid):
+    secret = config['registry'].settings['userid_hmac_secret']
+    digest = hmac_digest(secret, uid)
+    return str(uuid.UUID(digest[:32]))
+
+
+def process_account_events(config, queue_name, aws_region=None,
+                           queue_wait_time=20):
+    """Process account events from an SQS queue.
+
+    This function polls the specified SQS queue for account-related events,
+    processing each as it is found.  It polls indefinitely and does not return;
+    to interrupt execution you'll need to e.g. SIGINT the process.
+    """
+    logger.info("Processing account events from %s", queue_name)
+    try:
+        # Connect to the SQS queue.
+        # If no region is given, infer it from the instance metadata.
+        if aws_region is None:
+            logger.debug("Finding default region from instance metadata")
+            aws_region = ec2_metadata.region
+
+        logger.debug("Connecting to queue %r in %r", queue_name, aws_region)
+        sqs = boto3.resource('sqs', region_name=aws_region)
+        queue = sqs.get_queue_by_name(QueueName=queue_name)
+
+        # Poll for messages indefinitely.
+        # Use a wacky looping construct that can be mocked in tests.
+        for x in itertools.count():
+            msgs = queue.receive_messages(WaitTimeSeconds=queue_wait_time)
```

Comment:
Maybe you should sleep for a while (queue_wait_time probably) here before trying again.

Reply:
Alternatively you could consider https://pypi.org/project/pySqsListener/

Reply:
Passing `WaitTimeSeconds` ensures that the request doesn't return for up to `queue_wait_time` seconds, even if there are no messages. So doing `receive_messages` is both a `get` and a `sleep`.

pySqsListener is an interesting abstraction, but it only saves us about 5 lines of code. I think it might be more important to be consistent with the pattern used in the tokenserver code.